### PR TITLE
chore: udpate docs for the telemetry crate

### PR DESCRIPTION
**Pull request by praveen-influx**
Added more docs for the telemetry crate

### CODE REVIEW

**Comment by hiltontj** [comment]
Not sure why `cargo-audit` is failing, since I tried it locally on latest `main` and it passed 🤷 - but either way, I opened a `cargo update` PR here: https://github.com/influxdata/influxdb/pull/25433, so if that passes we can merge and then update this branch to get CI 🟢

File: influxdb3_telemetry/src/stats.rs [hunk 1]
```diff
@@ -75,6 +97,16 @@ impl<T: Default + Num + Copy + NumCast + PartialOrd> Stats<T> {
...
 }
 
+/// Generic function to calculate min/max/avg from another set of stats.
+/// This function works for all types of numbers (unsigned/signed/floats).
+/// It calculates min/max/avg by using already calculated min/max/avg for
+/// possibly a higher resolution.
+///
+/// For eg. the minimum number of
+/// lines written in a minute interval, is compared with other minute
+/// minute intervals sampled so far. If 1st sample's minimum was 20
```
Comment (hiltontj) [inline influxdb3_telemetry/src/stats.rs:107]: > For eg. the minimum number of lines written in a minute interval, is compared with other minute minute intervals sampled so far.

I think this sentence needs some wordsmithing.

**Review by hiltontj — COMMENTED** [review]
There is one sentence I found a bit confusing and commented on, and CI needs a fixin', but otherwise this is good.

**Comment by hiltontj** [comment]
https://github.com/influxdata/influxdb/pull/25433 is merged now, so hopefully bringing in `main` will fix CI here.

File: influxdb3_telemetry/src/sender.rs [hunk 2]
```diff
@@ -68,6 +70,8 @@ pub(crate) struct TelemetryPayload {
     pub parquet_row_count: u64,
 }
 
+/// This function runs in the background and if any call fails
+/// there is no retrying mechanism and it is ok to lose few samples
```
Comment (mgattozzi) [inline influxdb3_telemetry/src/sender.rs:74]:
```suggestion
/// there is no retrying mechanism and it is ok to lose a few samples
```

File: influxdb3_telemetry/src/stats.rs [hunk 3]
```diff
@@ -1,5 +1,17 @@
 use num::{Num, NumCast};
 
+/// This type is responsible for calculating stats in a rolling fashion.
+/// By rolling, it means that there is already some stats calculated
+/// which needs to be further aggregated. This is commonly the case when
+/// the sampling is done at a higher precision interval (say 1 minute) and
+/// then further aggregated (say 1 hour).
+///
+/// For example the number of lines written per hour is collected as new
+/// write requests come in. However the bucket [`crate::bucket::EventsBucket`]
```
Comment (mgattozzi) [inline influxdb3_telemetry/src/stats.rs:10]:
```suggestion
/// write requests come in. However, the bucket [`crate::bucket::EventsBucket`]
```

**Review by mgattozzi — APPROVED** [review]
These are really well documented @praveen-influx I love it! I added two easy to merge minor fixes, but for me this looks good.

File: influxdb3_telemetry/src/stats.rs (hunk 1 above)
Comment (praveen-influx) [reply influxdb3_telemetry/src/stats.rs]: @hiltontj I've merged this PR after updating the example, here is the link for the update from the other [PR](https://github.com/influxdata/influxdb/pull/25432), if you'd like to make any further suggestions please do it in that one 

https://github.com/influxdata/influxdb/blob/a8ce071a55e215688240d13a7a9c9772512ddfa7/influxdb3_telemetry/src/stats.rs#L105-L114

**Pull request merged**